google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java

```diff
@@ -3235,6 +3235,116 @@ public void testEnableAndDisableUniformBucketLevelAccessOnExistingBucket() throw
     }
   }
 
+  @Test
+  public void testUnspecifiedAndEnforcedPublicAccessPreventionOnBucket() throws Exception {
```

Comment:
nit: This test feels like 3 tests to me
1. When ENFORCED 
    * chaging bucket policy to enforced should fail
    * creating a public object should fail
2. When UNSPECIFIED 
    * chaging bucket policy to enforced should succeed
    * creating a public object should succeed
3. Updating UBLA should not modify PublicAccessPrevention